website/views.py

```diff
@@ -155,19 +155,47 @@ def add_unavailability():
     
     try:
         print(f"unavailable_start: {unavailable_start}, unavailable_end: {unavailable_end}")
-        # Convert string dates to datetime objects
-        unavailable_start_dt = datetime.fromisoformat(unavailable_start)
-        unavailable_end_dt = datetime.fromisoformat(unavailable_end)
+        unavailable_start_cleaned = unavailable_start.replace("Z", "")
+        unavailable_start_date = datetime.fromisoformat(unavailable_start_cleaned)
+        unavailable_end_cleaned = unavailable_end.replace("Z", "")
+        unavailable_end_date = datetime.fromisoformat(unavailable_end_cleaned)
 
-        # Create a new unavailability record
         new_unavailability = Unavailability(
             employeeID=current_user.employeeID,
-            unavailableStartTime=unavailable_start_dt,
-            unavailableEndTime=unavailable_end_dt
+            unavailableStartTime=unavailable_start_date,
+            unavailableEndTime=unavailable_end_date
         )
+        overlapping_intervals = (Unavailability.query
+                                 .filter(Unavailability.employeeID == current_user.employeeID)
+                                 .filter(Unavailability.unavailableEndTime > unavailable_start_date)
+                                 .filter(Unavailability.unavailableStartTime < unavailable_end_date)
+                                 .order_by(Unavailability.unavailableStartTime)
+                                 .all())
         
-        # Add to the database session and commit
-        db.session.add(new_unavailability)
+        
+        current_start = unavailable_start_date
```

Comment:
Logic looks fine. Might want to change some of the names so it makes more sense. Can be a bit hard to follow.

Reply:
or just give some comments.